### ISSUE TITLE
[REVIEW]: Manopt.jl: Optimization on Manifolds in Julia

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3866]
**Submitting author:** @kellertuer (<a href="http://orcid.org/0000-0001-8342-7218">Ronny Bergmann</a>)
**Repository:** <a href="https://github.com/JuliaManifolds/Manopt.jl" target ="_blank">https://github.com/JuliaManifolds/Manopt.jl</a>
**Version:** 0.3.17
**Editor:** @dpsanders
**Reviewer:** @krystophny, @sweichwald
**Archive:** <a href="https://doi.org/10.5281/zenodo.6019027" target="_blank">10.5281/zenodo.6019027</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b/status.svg)](https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b"><img src="https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b/status.svg)](https://joss.theoj.org/papers/803fd5cc2034643ea1476d45f9df669b)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JuliaManifolds/Manopt.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@krystophny & @sweichwald, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dpsanders know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @krystophny
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/JuliaManifolds/Manopt.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaManifolds/Manopt.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kellertuer) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @sweichwald
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/JuliaManifolds/Manopt.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaManifolds/Manopt.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kellertuer) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by kellertuer** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kellertuer** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @sweichwald, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @krystophny, please update us on how your review is going (this is an automated reminder).

**Comment by krystophny** [comment]
Hi everyone, sorry for the slowness - I will manage to finish until end of the week.

**Comment by krystophny** [comment]
@dpsanders @kellertuer I have completed my review of the `Manopt.jl` package and the software paper. Please forgive the long time it took, despite the optimism on finishing sooner. After a detailed analysis, I haven't found any obvious requirements to improve, so I can recommend publication in the present form.

In my opinion, software and paper are of high quality and up to standards in terms of scientific methods, design, documentation and testing. Compared to the existing related `Manopt` packages in Matlab and Python, `Manopt.jl` has been redesigned from ground up, runs on the high-performance Julia language, and implements novel algorithms in the field from the author's active field of research. I think the package looks highly useful, and could also be interesting to apply in my field of research (stellarator optimization).

**Comment by sweichwald** [comment]
@dpsanders @kellertuer I have now also completed my review.

Some minor comments on the software paper have been addressed (cf. https://github.com/JuliaManifolds/Manopt.jl/issues/96),
software design, development, documentation, and continuous integration follow best practices,
and, in my opinion, `Manopt.jl` is a useful quality piece of research software that I recommend for publication.

**Comment by kellertuer** [comment]
@dpsanders good morning and happy new year. May I ask what the next steps are here?

**Comment by dpsanders** [comment]
👋 @krystophny and @sweichwald: Many thanks for your reviews and comments; apologies for the delay here.

@kellertuer: Many apologies for the long delay. I'll proceed to check through the latest version of the paper.

**Comment by dpsanders** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
It looks like this paper includes an ORCID (0000-0003-0872-7098) that is probably not correct.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kellertuer** [comment]
Cool! Ah and sorry I somehow scrambled my orcid. Just fixed that.

**Comment by kellertuer** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kellertuer** [comment]
And carefully checking the paper (in the just generated pdf) I just found one type (a missing ` at the end of a code markup). But the rest seems fine.

**Comment by kellertuer** [comment]
Hm, anything I can help with here for the final steps?

**Comment by dpsanders** [comment]
@kellertuer I sent a PR with some minor edits for typos and grammar suggestions.

Could you please add a couple of references for the applications listed in the Statement of Need, and to the ManifoldsBase.jl package.

Otherwise this seems ready!

Final steps:
- Please archive it, e.g. at Zenodo, and paste the URL here.  Please make sure the title and author info in Zenodo are *exactly* the same as in the paper.
- Record here the version of the software corresponding to the archived version.

**Comment by kellertuer** [comment]
Oh, sorry for my impatience there, I was not expecting that you would also read it again :)

I just accepted and merged your patch.

Concerning the references – sure, I can add those later this week (hopefully tomorrow). Is it ok if I just link to ManifoldsBase.jl (and Manifolds.jl), for example do their package names as links to the repositories?

To the final steps
* I would merge the joss branch anyways and could just register a new version of Manopt.jl – I could also adapt the text to mention this version already. Would that fit in the sense of archiving the paper or should I archive the paper independently of the automated process (i.e. also _only_ the paper, so `paper.md` and `bibliography.bib`) ?
* sure then mentioning the software version here is not a problem.

**Comment by kellertuer** [comment]
I hope I finished most of your requests.

1. every application mentioned now also has a reference
2. I link to ManifoldsBase.jl on first occurrence (similar for Manifolds.jl) – but ManifoldsBase.jl does (not yet) have a zenodo DOI - we plan to do that with the next major release, which will also cover an own documentation (independent from Manifolds.jl) but this might still take a while.+

I already checked the result with whedon (the pre submit testy thing) and it looks finde, but I will trigger a build here as well.

So - is this ok? I am not sure what else to do with ManifoldsBase.jl

And – you wrote I should archive „it“ - just the paper? Or the whole software? The rules read like the software - but then the .zenodo file I have also covers everyone – even contributing minor functions – in said zenodo release.

However – I already prepared said version on the JOSS branch – merging that I will. tag 0.3.17 (which is now also the version mentioned in the paper.

So please give me a short feedback
1. ManifoldsBase.jl reference ok?
2. “just” 0.3.17 zenodo archive (automatic) ok or should I do a separate one just for the paper (the 0.3.17 will happen anyways)


For the whole software, we can surely do that by just merging the joss branch and doing a minor version bump (

**Comment by kellertuer** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
Thanks, the Zenodo archive I meant is for the software. So I think we're good, thanks.

**Comment by kellertuer** [comment]
Supert! I just merged the JOSS branch (it still exists of course to not confuse whedon) and registered that as a new version.

the new DOI including the paper is [10.5281/zenodo.6019027](https://doi.org/10.5281/zenodo.6019027), it includes the paper as well as a script file for the example done in the paper – and this example also recreates the figure if run.

Thanks for all your help :)

**Comment by dpsanders** [comment]
@whedon set archive as https://doi.org/10.5281/zenodo.6019027

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by dpsanders** [comment]
@whedon set https://doi.org/10.5281/zenodo.6019027 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.6019027" target="_blank">10.5281/zenodo.6019027</a> is the archive.

**Comment by dpsanders** [comment]
@whedon set 0.3.17 as version

**Comment by whedon** [comment]
OK. 0.3.17 is the version.

**Comment by dpsanders** [comment]
@whedon dry-accept

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by dpsanders** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by dpsanders** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1515/9781400830244 is OK
- 10.1137/140953393 is OK
- 10.1007/s10208-020-09486-5 is OK
- 10.1137/15M1052858 is OK
- 10.1137/140955483 is OK
- 10.1137/15M101988X is OK
- 10.3389/fams.2018.00059 is OK
- 10.3934/ipi.2016001 is OK
- 10.1007/978-3-642-15461-4_2 is OK
- 10.1145/3218822 is OK
- 10.1137/140955483 is OK
- 10.1007/s10208-005-0179-9 is OK
- 10.1002/cpa.3160300502 is OK
- 10.1007/978-3-030-80209-7_82 is OK
- 10.1007/s10851-018-0827-8 is OK
- 10.1137/120867172 is OK
- 10.1109/ICCV.2013.366 is OK
- 10.1016/j.ultramic.2011.08.002 is OK
- 10.1137/110845768 is OK
- 10.1007/s10851-018-0840-y is OK
- 10.1007/s10589-021-00328-w is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2935

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2935, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi @kellertuer! a few items:

1. version and zenodo look good ✅ 
2. paper:
- statement of need, first paragraph: you have a list of references but each are in their own set of parentheses. Put the brackets on the outside of the list, with ";" between, to get one set of parentheses on the outside. [docs](@whedon generate pdf from branch joss-paper)
- functionality, paragraph 2: "see e.g. (..." but the reference should not be parenthetical. Just remove the [] around the reference.
- other references cited inline I see should not be parenthetical sometimes — remove the brackets when they should be inline.
- Preserve capitalization in your references by putting {} around characters in your .bib file. For example, Absil et al has 
"riemannian" not capitalized. Check the other references too.

**Comment by kellertuer** [comment]
Thanks for the tipps 

* I actually only learned about `;` in cites yesterday and had hoped I changed all places – seem to have missed the one in the fist paragraph - thanks, fixed that
* oh, I was not aware of the inline cite. I adapted the mention cite and all other places where the text was using online cites – they even work with ; - nice!
* sorry for the capitalisation errors, I was a little uncareful yesterday I think.
* While going through capitalisation I even notices I had one reference double-included and reduced that to only once having that in the bib-file.

**Comment by kellertuer** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kellertuer** [comment]
Hm, that is strange – one point might be, that I merged the joss branch into master before doing the zenodo mentioned above, I can just add them to master as well, but Whedon should take the joss branch as we both called it – but I can also not see the changes in the current article proofs.

**Comment by kthyng** [comment]
Yeah Zenodo shouldn't impact! Weird. I see you are doing some stuff. Let's try generating again until we can see the changes.

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03866/joss.03866/10.21105.joss.03866.pdf) :page_facing_up: :point_left:

**Comment by kellertuer** [comment]
Maybe Whedon was confused, now the changes are also in the PDF :)

**Comment by kthyng** [comment]
Ok, whatever the problem was, seems better now! We can proceed.

**Comment by kthyng** [comment]
@whedon accept from branch joss deposit=true

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by kthyng** [comment]
@whedon recommend-accept from branch joss

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1515/9781400830244 is OK
- 10.1137/140953393 is OK
- 10.1007/s10208-020-09486-5 is OK
- 10.1137/15M1052858 is OK
- 10.1137/15M101988X is OK
- 10.3389/fams.2018.00059 is OK
- 10.3934/ipi.2016001 is OK
- 10.1007/978-3-642-15461-4_2 is OK
- 10.1145/3218822 is OK
- 10.1137/140955483 is OK
- 10.1007/s10208-005-0179-9 is OK
- 10.1002/cpa.3160300502 is OK
- 10.1007/978-3-030-80209-7_82 is OK
- 10.1007/s10851-018-0827-8 is OK
- 10.1137/120867172 is OK
- 10.1109/ICCV.2013.366 is OK
- 10.1016/j.ultramic.2011.08.002 is OK
- 10.1137/110845768 is OK
- 10.1007/s10851-018-0840-y is OK
- 10.1007/s10589-021-00328-w is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2941

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2941, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss 
```

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1491890401858695168) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2942
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03866](https://doi.org/10.21105/joss.03866)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new publication @kellertuer! Many thanks to editor @dpsanders and reviewers @krystophny and @sweichwald for your time, hard work, and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03866/status.svg)](https://doi.org/10.21105/joss.03866)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03866">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03866/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03866/status.svg
   :target: https://doi.org/10.21105/joss.03866
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03866/status.svg)](https://doi.org/10.21105/joss.03866)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by dpsanders** [comment]
Congratulations on your paper, @kellertuer 🎉 ! Many thanks to @krystophny and @sweichwald for your expertise and reviews, and to @kthyng for help with the final push.
